code42/py42
# Bugfix/handle restore strategy

### Description of Change ###

Adding ability to use the `file_location` parameter for restoring to original location during a `archive.stream_to_device()`.
The constants are available in `py42.services.storage.restore.PushRestoreLocation`

Also, you will get a custom exception, `Py42BadRestoreError`, when creating the push restore fails.. 

AND if the GUIDs differ and you are using the `ORIGINAL_LOCATION` location strategy, you will get an additional warning message describing that, as that may be related to the error.

### Issues Resolved ###
Push restore feature, no github issue.

- closes #

### Testing Procedure ###
*Verify you can use the file_location param and that it shows up in DEBUG request logs.

* do a Push restore when you know the devices have differing paths and use the ORIGINAL_LOCATION strategy and get the custom exception.

### PR Checklist ###
Did you remember to do the below?

- [n/a] Add an entry to CHANGELOG.md describing this change
